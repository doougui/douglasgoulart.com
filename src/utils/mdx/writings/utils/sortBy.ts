import { Writing } from 'types/Writing';
import { parseToDate } from 'utils/time/parseToDate';
import { SortTypesKeys } from '../types';

function byNew(data: Writing[]) {
  return data.sort((a, b) => {
    const aAsDate = parseToDate(a.publishedAt);
    const bAsDate = parseToDate(b.publishedAt);
    return Number(bAsDate) - Number(aAsDate);
  });
}

function byName(data: Writing[]) {
  return data.sort((a, b) => (a.title > b.title ? 1 : -1));
}

export function sortBy(data: Writing[], type: SortTypesKeys): Writing[] {
  const sortTypes: { [key in SortTypesKeys]: () => Writing[] } = {
    new: () => byNew(data),
    name: () => byName(data),
  };

  return sortTypes[type]();
}
