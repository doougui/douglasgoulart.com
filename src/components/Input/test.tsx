import { render, screen, waitFor } from 'utils/tests';
import userEvent from '@testing-library/user-event';

import { Input } from '.';

describe('<Input />', () => {
  it('should render the input', () => {
    const { container } = render(<Input placeholder="input" />);

    expect(screen.getByPlaceholderText('input')).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render with error', () => {
    render(<Input name="test" hasError />);

    expect(screen.getByRole('textbox')).toHaveStyleRule(
      'border',
      '1px solid var(--color-danger)',
    );
  });

  it('changes its value when typing', async () => {
    const onInputChange = jest.fn();
    render(<Input onInputChange={onInputChange} />);

    const input = screen.getByRole('textbox');
    const text = 'This is my new text';
    userEvent.type(input, text);

    await waitFor(() => {
      expect(input).toHaveValue(text);
      expect(onInputChange).toHaveBeenCalledTimes(text.length);
    });
    expect(onInputChange).toHaveBeenCalledWith(text);
  });

  it('does not change its value when disabled', async () => {
    const onInputChange = jest.fn();
    render(<Input onInputChange={onInputChange} disabled />);

    const input = screen.getByRole('textbox');
    expect(input).toBeDisabled();

    const text = 'This is my new text';
    userEvent.type(input, text);

    await waitFor(() => {
      expect(input).not.toHaveValue(text);
    });
    expect(onInputChange).not.toHaveBeenCalled();
  });
});
