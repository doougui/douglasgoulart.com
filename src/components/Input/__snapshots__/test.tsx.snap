// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should render the input 1`] = `
.c0 {
  all: unset;
  width: 100%;
  font-size: 1.6rem;
  background: #2D2D2F;
  padding: 1.2rem 1.5rem;
  border-radius: 0.4rem;
  color: #EDEDEF;
}

.c0::-webkit-input-placeholder {
  color: #888888;
}

.c0::-moz-placeholder {
  color: #888888;
}

.c0:-ms-input-placeholder {
  color: #888888;
}

.c0::placeholder {
  color: #888888;
}

<input
  class="c0"
  placeholder="input"
  value=""
/>
`;
